sklearn/base.py

```diff
@@ -290,10 +290,15 @@ def __repr__(self):
                                                offset=len(class_name),),)
 
     def __getstate__(self):
+        try:
+            state = super(BaseEstimator, self).__getstate__()
+        except AttributeError:
+            state = self.__dict__.copy()
+
         if type(self).__module__.startswith('sklearn.'):
-            return dict(self.__dict__.items(), _sklearn_version=__version__)
+            return dict(state.items(), _sklearn_version=__version__)
```

Comment:
Firstly that is not a problem for this PR. Secondly, we do not issue the warning if out of sklearn because such an estimator is likely to be versioned differently.